agents/labjack/labjack_agent.py

```diff
@@ -342,6 +349,104 @@ def stop_acq(self, session, params=None):
         else:
             return False, 'acq is not currently running'
 
+    def start_acq_reg(self, session, params=None):
+        """
+        Task to start data acquisition when you want to read out
+        non-standard registers. In particular the custom registers
+        labjack has built for reading out thermocouples. Maximum is
+        about 2.5 Hz but is set by the register read time which is 
+        estimated at the beginning of the acq_reg setup step.
+
+        Args:
+            sampling_frequency (float):
+                Sampling frequency for data collection. Defaults to 2.5 Hz
+                Maximum set by the register read time. Will reset to lower
+                rate if faster than possible read time.
+
+        """
+        if params is None:
+            params = {}
+        #Determine the read time latency to set the max allowable 
+        #sampling rate by reading the register 100 times in a row 
+        #and recording the time it takes to read each time. Then
+        #setting the max sample rate to be 50mS greater than the median
+        #of the time it took to read.
+        num_chs = len(self.chs)
+        times = []
+        for i in range(100):
+            times.append(time.time())
+            ljm.eReadNames(self.handle,num_chs,self.chs)[0]
+        read_dt = np.round(np.median(np.diff(times)),2)+0.05
+        max_fsamp = 1/read_dt
```

Comment:
This at most is ~2.5 Hz?

Reply:
Made it more clear that it is hard fixed at an upper bound of 2.5 Hz by adding in a line `max_fsamp = min(2.5,1/read_dt)`